### PR TITLE
Skip whitespaces after opening { in npy header

### PR DESCRIPTION
**Pull request by Luthaf**
Fixes https://github.com/lab-cosmo/equistore/pull/238#issuecomment-1498040918

This is a follow up to #240, which broke reading files we just wrote

### CODE REVIEW

File: equistore-core/src/io/npy_header.rs [hunk 1]
```diff
@@ -277,13 +277,14 @@ impl std::fmt::Display for DataType {
     }
 }
 
-#[derive(Clone, Debug)]
+#[derive(Clone, Debug, PartialEq)]
```
Comment (agoscinski) [inline equistore-core/src/io/npy_header.rs:280]: Why we need this for this fix?

**Review by agoscinski — APPROVED** [review]
It works. there is just two questions but I already approved


Header of npz file changed from
```
$ head tmp.npz
PK!(keys.npyNUMPYv{ 'descr': [('key_1', '<i4'), ('key_2', '<i4'), ], 'fortran_order': False, 'shape': (4, ) }                          
PK!(dblocks/0/values/data.npyNUMPYv{ 'descr': '<f8', 'fortran_order': False, 'shape': (3, 1, 1, ) }                                                     
???PK!(locks/0/values/samples.npyNUMPYv{ 'descr': [('samples', '<i4'), ], 'fortran_order': False, 'shape': (3, ) }                                          
PK!(Z4 blocks/0/values/components/0.npyNUMPYv{ 'descr': [('components', '<i4'), ], 'fortran_order': False, 'shape': (1, ) }                                       
PK!(2dblocks/0/values/properties.npyNUMPYv{ 'descr': [('properties', '<i4'), ], 'fortran_order': False, 'shape': (1, ) }                                       
PK!~%blocks/0/gradients/parameter/data.npyNUMPYv{ 'descr': '<f8', 'fortran_order': False, 'shape': (2, 1, 1, ) }                                                     
&@&@PK!(bёy(blocks/0/gradients/parameter/samples.npyNUMPYv{ 'descr': [('sample', '<i4'), ('parameter', '<i4'), ], 'fortran_order': False, 'shape': (2, ) }                     
PK!(Z4-blocks/0/gradients/parameter/components/0.npyNUMPYv{ 'descr': [('components', '<i4'), ], 'fortran_order': False, 'shape': (1, ) }                                       
PK!(-blocks/1/values/data.npyNUMPYv{ 'descr': '<f8', 'fortran_order': False, 'shape': (3, 1, 3, ) }                                                     
@@@@@@@@@PK!(̟locks/1/values/samples.npyNUMPYv{ 'descr': [('samples', '<i4'), ], 'fortran_order': False, 'shape': (3, ) }                                          
```
(Note the whitespaces in the shape)
 To

```
PK!('keys.npyNUMPYv{'descr': [('key_1', '<i4'), ('key_2', '<i4')], 'fortran_order': False, 'shape': (4,)}                               
PK!T>blocks/0/values/data.npyNUMPYv{'descr': '<f8', 'fortran_order': False, 'shape': (3, 1, 1)}                                                         
???PK!(1locks/0/values/samples.npyNUMPYv{'descr': [('samples', '<i4')], 'fortran_order': False, 'shape': (3,)}                                               
PK!(_x blocks/0/values/components/0.npyNUMPYv{'descr': [('components', '<i4')], 'fortran_order': False, 'shape': (1,)}                                            
PK!(-;blocks/0/values/properties.npyNUMPYv{'descr': [('properties', '<i4')], 'fortran_order': False, 'shape': (1,)}                                            
PK!(0%blocks/0/gradients/parameter/data.npyNUMPYv{'descr': '<f8', 'fortran_order': False, 'shape': (2, 1, 1)}                                                         
&@&@PK!((blocks/0/gradients/parameter/samples.npyNUMPYv{'descr': [('sample', '<i4'), ('parameter', '<i4')], 'fortran_order': False, 'shape': (2,)}                          
PK!(_x-blocks/0/gradients/parameter/components/0.npyNUMPYv{'descr': [('components', '<i4')], 'fortran_order': False, 'shape': (1,)}                                            
PK!(s8blocks/1/values/data.npyNUMPYv{'descr': '<f8', 'fortran_order': False, 'shape': (3, 1, 3)}                                                         
@@@@@@@@@PK!(Yilocks/1/values/samples.npyNUMPYv{'descr': [('samples', '<i4')], 'fortran_order': False, 'shape': (3,)}                                               
```

**Comment by Luthaf** [comment]
> Header of npz file changed from

Changed between master and this PR or from before #240 and this PR? Anyway, both are valid tuple literal, and the code should be able to parse both.

**Comment by PicoCentauri** [comment]
Can we merge this? My workflow is broken currently?

File: equistore-core/src/io/npy_header.rs [hunk 2]
```diff
@@ -661,5 +664,9 @@ mod tests {
             shape: vec![3],
         };
         assert_eq!(header.to_dict_literal(), "{ 'descr': '<f8', 'fortran_order': True, 'shape': (3, ) }");
+
+        // check round trip
+        let parsed = Header::from_str(&header.to_dict_literal()).unwrap();
```
Comment (Luthaf) [inline equistore-core/src/io/npy_header.rs:669]: @PicoCentauri this is the test checking that we can read again the header we produced, which is the root cause of the bug

Comment (PicoCentauri) [reply equistore-core/src/io/npy_header.rs:669]: ahh okay then I am happy and we can merge!

**Review by PicoCentauri — APPROVED** [review]

**Pull request merged**